lib/ably/realtime/connection.rb

```diff
@@ -66,7 +66,7 @@ class Connection
       ensure_state_machine_emits 'Ably::Models::ConnectionStateChange'
 
       # Expected format for a connection recover key
-      RECOVER_REGEX = /^(?<recover>[\w!-]+):(?<connection_serial>\-?\w+)$/
+      RECOVER_REGEX = /^(?<recover>[\w!-]+):(?<connection_serial>\-?\w+):(?<msg_serial>\-?\d+)$/
```

Comment:
Yeh, I didn't like it either to be honest.  I'll assume it's therefore just a string in three parts delimited by a `:`, otherwise it will treat it as invalid.